src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -140,6 +140,8 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
         foreach ($tokens as $token) {
             if ('variable' === $token[0]) {
                 if (!$optional || !array_key_exists($token[3], $defaults) || null !== $mergedParams[$token[3]] && (string) $mergedParams[$token[3]] !== (string) $defaults[$token[3]]) {
+                    // remove look-around
+                    $token[2] = preg_replace('/\(\?(?:=|<=|!|<!)((?:[^()\\\\]+|\\\\.|\((?1)\))*)\)/', '', $token[2]);
```

Comment:
This should only be done when `null !== $this->strictRequirements`. Also the same look-around problem applies to the host-tokens.

Reply:
updated, thanks!